File: backend/src/main/java/com/medicaloffice/medicalofficemanager/auth/AuthService.java
package com.medicaloffice.medicalofficemanager.auth;

import com.medicaloffice.medicalofficemanager.auth.dto.AuthResponse;
import com.medicaloffice.medicalofficemanager.auth.dto.LoginRequest;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Service;

@Service
public class AuthService {
    private static final Logger log = LoggerFactory.getLogger(AuthService.class);

    private final AuthenticationManager authenticationManager;
    private final JwtService jwtService;

    public AuthService(AuthenticationManager authenticationManager,
                       JwtService jwtService) {
        this.authenticationManager = authenticationManager;
        this.jwtService = jwtService;
    }

    public AuthResponse login(LoginRequest request) {
        Authentication authentication = authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(
                        request.getUsername(),
                        request.getPassword()
                )
        );

        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
        String token = jwtService.generateToken(userDetails);

        String role = "";
        if (userDetails instanceof CustomUserDetails customUserDetails) {
            role = customUserDetails.getRole();
        }

        log.info("User '{}' logged in successfully", request.getUsername());
        return new AuthResponse(token, role);
    }
}

File: backend/src/main/java/com/medicaloffice/medicalofficemanager/auth/AuthService.kt
package com.medicaloffice.medicalofficemanager.auth

import com.medicaloffice.medicalofficemanager.auth.dto.AuthResponse
import com.medicaloffice.medicalofficemanager.auth.dto.LoginRequest
import org.slf4j.LoggerFactory
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Service

@Service
class AuthService(
    private val authenticationManager: AuthenticationManager,
    private val jwtService: JwtService,
) {
    private val log = LoggerFactory.getLogger(AuthService::class.java)

    fun login(request: LoginRequest): AuthResponse {
        val authentication = authenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(
                request.username(),
                request.password()
            )
        )

        val userDetails = authentication.principal as UserDetails
        val token = jwtService.generateToken(userDetails)

        val role = (userDetails as? CustomUserDetails)?.role
            ?: throw IllegalStateException("Expected CustomUserDetails but got ${userDetails::class.simpleName}")

        log.info("User '{}' logged in successfully", request.username())
        return AuthResponse(token, role)
    }
}
